### PR TITLE
JAMES-2991 Manage deletion in ComputeMessageFastViewProjectionListener

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/james-project/issues/4867

### CODE REVIEW

File: server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/ComputeMessageFastViewProjectionListener.java [hunk 1]
```diff
@@ -99,4 +109,14 @@ private Mono<Void> handleAddedEvent(Added addedEvent, MailboxSession session) {
     MessageFastViewPrecomputedProperties computeFastViewPrecomputedProperties(MessageResult messageResult) throws MailboxException, IOException {
         return messageFastViewPrecomputedPropertiesFactory.from(messageResult);
     }
+
+    private Mono<Void> handedExpungedEvent(Expunged expunged, MailboxSession session) {
+        ImmutableSet<MessageId> expungedMessageIds = expunged.getMessageIds();
+        return Mono.from(messageIdManager.accessibleMessagesReactive(expungedMessageIds, session))
+            .map(accessibleMessageIds -> CollectionUtils.subtract(expungedMessageIds, accessibleMessageIds))
+            .flatMapMany(Flux::fromIterable)
```
Comment (chibenwa) [inline server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/ComputeMessageFastViewProjectionListener.java:117]:
```suggestion
            .flatMapIterable(accessibleMessageIds -> CollectionUtils.subtract(expungedMessageIds, accessibleMessageIds))
```

?

**Review by chibenwa — APPROVED** [review]
Otherwise looks good to me.

**Review by chibenwa — APPROVED** [review]

File: server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/ComputeMessageFastViewProjectionListener.java [hunk 2]
```diff
@@ -99,4 +109,13 @@ private Mono<Void> handleAddedEvent(Added addedEvent, MailboxSession session) {
     MessageFastViewPrecomputedProperties computeFastViewPrecomputedProperties(MessageResult messageResult) throws MailboxException, IOException {
         return messageFastViewPrecomputedPropertiesFactory.from(messageResult);
     }
+
+    private Mono<Void> handedExpungedEvent(Expunged expunged, MailboxSession session) {
```
Comment (Arsnael) [inline server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/ComputeMessageFastViewProjectionListener.java:113]:
```suggestion
    private Mono<Void> handleExpungedEvent(Expunged expunged, MailboxSession session) {
```

**Review by Arsnael — COMMENTED** [review]
Just a typo, fine otherwise :)

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by hungphan227 — APPROVED** [review]

**Pull request merged**